### PR TITLE
Improve input tensor structure validation algorithm

### PR DESCRIPTION
**Pull request by anight**
Existing code that does input tensor validation is unable to check slices sizes properly.
Consider this input:
`[][][]float32{{{1, 2}, {3, 4}}, {{1}, {3}}, {{1, 2, 3}, {2, 3, 4}}}`
NewTensor() currently treat this input as valid one.
This patch fixes this behaviour and also simplifies the check by removing unnecessary for cycles.

### CODE REVIEW

File: tensorflow/go/tensor.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ func NewTensor(value interface{}) (*Tensor, error) {
 	raw := tensorData(t.c)
 	buf := bytes.NewBuffer(raw[:0:len(raw)])
 	if dataType != String {
-		if err := encodeTensor(buf, val); err != nil {
+		if err := encodeTensor(buf, val, shape); err != nil {
```
Comment (jhseu) [inline tensorflow/go/tensor.go]: This means that we no longer do the check on string types, right?

Comment (anight) [reply tensorflow/go/tensor.go]: You are absolutely right, I missed this point.
Here is a fix for strings as well

**Review by jhseu — APPROVED** [review]

**Comment by gunan** [comment]
Jenkins, test this please.

**Comment by sb2nov** [comment]
Jenkins, test this please.

**Comment by sb2nov** [comment]
Jenkins, test this please.

**Pull request merged**